internal/registry/registry.go

```diff
@@ -247,3 +256,14 @@ func (r *registry) getImageDigestFromMultiImage(manifestListStream []byte) (stri
 	}
 	return "", fmt.Errorf("Failed to find manifest for architecture %s", arch)
 }
+
+func (r *registry) getRegistryAuth(secret *corev1.LocalObjectReference, namespace string) auth.RegistryAuthGetter {
```

Comment:
What will we do if we add another authenticator (or replace that one) with one that isn't based on a secret?